aiida_common_workflows/cli/plot.py

```diff
@@ -79,18 +87,23 @@ def cmd_plot_eos(node, precisions, print_table, output_file):
 
 
 @cmd_plot.command('dissociation-curve')
-@arguments.NODE()
+@arguments.WORKFLOW()
 @options.PRECISIONS()
 @options.PRINT_TABLE()
 @options.OUTPUT_FILE()
-def cmd_plot_dissociation_curve(node, precisions, print_table, output_file):
+def cmd_plot_dissociation_curve(workflow, precisions, print_table, output_file):
     """Plot the results from a `DissociationCurveWorkChain`."""
     # pylint: disable=too-many-locals
     from tabulate import tabulate
     from aiida.common import LinkType
     from aiida_common_workflows.common.visualization.dissociation import get_dissociation_plot
 
-    outputs = node.get_outgoing(link_type=LinkType.RETURN).nested()
+    if workflow.process_class is not DissociationCurveWorkChain:
+        echo.echo_critical(f'Provided workflow with PK {workflow.pk} does not represent a DissociationCurveWorkChain.')
+    if workflow.exit_status != 0:
```

Comment:
This is fine for now, but it might be possible in the future that there will be a non-zero exit status that still has valid outputs that can be plotted. Not sure how to deal with this then. Maybe at that point we will swap this check for simply checking whether the required output nodes are present as a more direct measure

Reply:
Should I maybe add a comment on this in the code? Or open an issue to keep track of this?

Reply:
Well, the _best_ would be to already do the change now :smirk: It is just basically checking that the correct outputs exist (not checking their contents, because that will be done by the actual `get_plot_*` function. But otherwise I would add a comment with this suggestion.

Reply:
Your ability to slip more work into my PR's is truly uncanny. 😅 Will fix it here!